helix-lsp/src/lib.rs

```diff
@@ -356,7 +356,13 @@ impl Registry {
                 let NewClientResult(client, incoming) = start_client(id, language_config, config)?;
                 self.incoming.push(UnboundedReceiverStream::new(incoming));
 
-                entry.insert((id, client.clone()));
+                let (_, old_client) = entry.insert((id, client.clone()));
+
+                tokio::spawn(async move {
+                    if let Err(e) = old_client.shutdown_and_exit().await {
```

Comment:
I wonder if we should also attempt to kill the child process ourselves as well in case this fails?

Reply:
It looks like that method does not attempt to do anything with the child process, so if the LS fails to kill itself after the `shutdown` and `exit` methods, the process may linger. Maybe that would be an appropriate place to try to kill the process though.